spring-integration-amqp/src/test/java/org/springframework/integration/amqp/outbound/RabbitStreamMessageHandlerTests.java

```diff
@@ -117,4 +125,108 @@ void sendNative() throws InterruptedException {
 		streamTemplate.close();
 	}
 
+	@Test
+	void errorChanelAsync() {
+		Environment env = Environment.builder()
+				.lazyInitialization(true)
+				.port(RabbitTestContainer.streamPort())
+				.build();
+		try {
+			env.deleteStream("stream.stream");
+		} catch (Exception e) {
+		}
+		env.streamCreator().stream("stream.stream").create();
+		RabbitStreamTemplate streamTemplate = new RabbitStreamTemplate(env, "stream.stream");
+
+		MessageChannel errorChannel = Mockito.mock(MessageChannel.class);
```

Comment:
This channel for testing could be a `QueueChannel` with very convenient `receive(timeout)` API.
So, anything you are mocking here and verify is fully redundant.